cmd/main.go

```diff
@@ -85,27 +136,45 @@ func main() {
 	fmt.Printf("blackholing enabled: %v\n", blackholing)
 	for {
 		fmt.Println(time.Now())
-		_ = netconfig.InitLoggingChain("4")
-		_ = netconfig.InitLoggingChain("6")
+		err := netconfig.InitLoggingChain("4")
+		if err != nil {
+			fmt.Printf("Error initializing ipv4 logging chain: %v\n", err)
+			os.Exit(1)
+		}
+		err = netconfig.InitLoggingChain("6")
+		if err != nil {
+			fmt.Printf("Error initializing ipv6 logging chain: %v\n", err)
+			os.Exit(1)
+		}
 		if blackholing {
 			err := netconfig.InitDummyDevice()
 			if err != nil {
 				fmt.Printf("Error initializing dummy device: %v", err)
 				os.Exit(1)
 			}
-			fmt.Printf("Updating blackhole routes...")
+			fmt.Println("Updating blackhole routes...")
 			err = updateBlackholeRoutes(ipV4List, ipV6List)
 			if err != nil {
 				fmt.Fprintf(os.Stderr, "Error updating blackhole routes: %v\n", err)
 				os.Exit(1)
 			}
+			err = cleanupFirewall()
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "Error cleaning up iptables: %v\n", err)
+				os.Exit(1)
+			}
 		} else {
 			fmt.Println("Updating iptables rules ...")
 			err := updateFirewall(blockIngress, ipV4List, ipV6List)
 			if err != nil {
 				fmt.Fprintf(os.Stderr, "Error updating iptables: %v\n", err)
 				os.Exit(1)
 			}
+			err = cleanupBlackholeRoutes()
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "Error cleaning up blackhole routes: %v\n", err)
+				os.Exit(1)
+			}
```

Comment:
The cleanup code is now executed in every loop iteration. Is this intended?
I assume we only need to clean-up once. In case we would need to clean-up regularly it would be likely that two instances would fight each other.
Do you envision another scenario/reason why this should be executed on each iteration?

Reply:
I didn't want to introduce additional state and relied on every operation being idempotent. I treat old rules not existing as part of the desired state which is to be constantly checked for, just like new rules that should exist.

Reply:
Ok, let's hope this works as expected and we do not see random/continuous crashes due to this.